kubernetes-extension/src/test/java/org/ballerinax/kubernetes/handlers/IstioGatewayGeneratorTests.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinax.kubernetes.handlers;
+
+import org.apache.commons.io.FileUtils;
+import org.ballerinax.kubernetes.exceptions.KubernetesPluginException;
+import org.ballerinax.kubernetes.handlers.istio.IstioGatewayHandler;
+import org.ballerinax.kubernetes.models.KubernetesContext;
+import org.ballerinax.kubernetes.models.istio.IstioGatewayModel;
+import org.ballerinax.kubernetes.models.istio.IstioPortModel;
+import org.ballerinax.kubernetes.models.istio.IstioServerModel;
+import org.junit.Assert;
+import org.junit.Test;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.ballerinax.kubernetes.KubernetesConstants.ISTIO_GATEWAY_FILE_POSTFIX;
+import static org.ballerinax.kubernetes.KubernetesConstants.YAML;
+
+/**
+ * Unit test cases for istio gateway models.
+ *
+ * @since 0.983.0
+ */
+public class IstioGatewayGeneratorTests {
+    @Test
+    public void testSimpleGateway() {
+        IstioGatewayModel istioGatewayModel = new IstioGatewayModel();
+        istioGatewayModel.setName("my-gateway");
+        istioGatewayModel.setNamespace("ballerina");
+        
+        Map<String, String> selectors = new LinkedHashMap<>();
+        selectors.put("app", "my-gatweway-controller");
```

Comment:
Shall we use `KubernetesConstants.KUBERNETES_SELECTOR_KEY` here?

Reply:
Fixed with https://github.com/ballerinax/kubernetes/pull/220/commits/5f8887c1388816ebe18d4066c627a7d43a88bea9